test/vtgatev3_test.py

```diff
@@ -948,12 +1005,55 @@ def test_main_seq(self):
     
     # Now test direct calls to sequence.
     result = self.execute_on_master(
-        vtgate_conn, "select next 1 values from vt_main_seq", {})
+        vtgate_conn, 'select next 1 values from vt_main_seq', {})
     self.assertEqual(
         result,
         ([(5,)], 1, 0,
          [('nextval', self.int_type)]))
 
+  def test_upsert(self):
+    vtgate_conn = get_connection()
+
+    # Create lookup entries for primary vindexx:
```

Comment:
I'm probably confused but shuldn't these entries be created by the insert of the rows on the main table? And if you're creating them explicitly, then I would have expected the second insert to fail since the vindex rows exist (and you're not using ignore).